turbo/snapshotsync/block_snapshots.go

```diff
@@ -1621,19 +1653,38 @@ RETRY:
 			}
 
 			isSystemTx := len(word) == 0
+			var isBorTx bool
 			if isSystemTx { // system-txs hash:pad32(txnID)
 				binary.BigEndian.PutUint64(slot.IDHash[:], firstTxID+i)
+			} else if borCfg.IsBor && blockNum%borCfg.Sprint == 0 && rawdb.HasBorReceipts(tx, blockNum) {
+				if _, isBorTx, err = parseCtx.ParseTransaction(word[1+20:], 0, &slot, make([]byte, 20), true /* hasEnvelope */, nil /* validateHash */); err != nil {
+					return fmt.Errorf("ParseTransaction: %w, blockNum: %d, i: %d", err, blockNum, i)
+				}
 			} else {
-				if _, err := parseCtx.ParseTransaction(word[1+20:], 0, &slot, nil, true /* hasEnvelope */, nil); err != nil {
+				if _, _, err = parseCtx.ParseTransaction(word[1+20:], 0, &slot, nil, true /* hasEnvelope */, nil /* validateHash */); err != nil {
 					return fmt.Errorf("ParseTransaction: %w, blockNum: %d, i: %d", err, blockNum, i)
 				}
 			}
 
-			if err := txnHashIdx.AddKey(slot.IDHash[:], offset); err != nil {
-				return err
-			}
-			if err := txnHash2BlockNumIdx.AddKey(slot.IDHash[:], blockNum); err != nil {
-				return err
+			if isBorTx {
+				header := rawdb.ReadHeaderByNumber(tx, blockNum)
```

Comment:
- to get header hash can use `ReadCanonicalHash`
- but Snapshots Indexing happening after downloading of snapshots - db is empty in that time.

Reply:
see how `bodyGetter` used instead of db in this method